packages/graphql-validate-fixtures/src/validate.ts

```diff
@@ -37,106 +82,106 @@ export interface Validation {
 
 const OPERATION_MARKER = '@operation';
 
-export function validateFixtureAgainstSchema(
-  fixture: Fixture,
-  schema: GraphQLSchema,
-): Validation {
-  const queryType = schema.getQueryType();
-  const mutationType = schema.getMutationType();
-  const {content, path} = fixture;
+function normalizeOperationName(operationName: string): string;
+function normalizeOperationName(
+  operationName: string | undefined,
+): string | undefined;
+function normalizeOperationName(operationName: string | undefined) {
+  return operationName
+    ? operationName.replace(/(Query|Mutation|Subscription)$/i, '')
+    : undefined;
+}
 
-  return {
-    fixturePath: path,
-    validationErrors: Object.keys(content).reduce((allErrors: Error[], key) => {
-      let rootType: GraphQLObjectType;
-      const keyPath = key;
+export function getOperationNames(fixture: Fixture): string[] {
+  const fixtureDirectoryName = basename(dirname(fixture.path));
+  const operationMarkerName: string | undefined =
+    fixture.content[OPERATION_MARKER];
+
+  return Array.from(
+    new Set([
+      fixtureDirectoryName,
+      operationMarkerName,
+      normalizeOperationName(fixtureDirectoryName),
+      normalizeOperationName(operationMarkerName),
+    ]),
+  ).filter(
+    (operationName): operationName is string =>
+      typeof operationName === 'string',
+  );
+}
 
-      if (key === OPERATION_MARKER) {
-        return allErrors;
-      }
+export interface FoundOperation {
```

Comment:
Can you please pull this to the top of the file?

Reply:
Also, this type confuses me; it has a single operation but also a bunch of operations?